.github/workflows/cmake.yml

```
@@ -0,0 +1,90 @@
+# inspired by "build GDAL library and to run test with CMake"
+# from https://github.com/OSGeo/gdal/blob/master/.github/workflows/cmake_builds.yml
+
+---
+name: CMake
+
+on:
+    push:
+        paths-ignore:
+            - 'doc/**'
+        branches-ignore:
+            - 'backport**'
+    pull_request:
+        paths-ignore:
+            - 'doc/**'
```

```suggestion
  push:
    paths-ignore:
      - 'doc/**'
    branches:
      - main
      - releasebranch_*
  pull_request:
    paths-ignore:
      - 'doc/**'
```

Uniformise with our other workflows, and prevents the double run (push and PR) that I saw on my side.

It uses the two space indent, as the complete file will someday be formatted near the end of the PR.

---

Can we mix 4 and 2 indentations?

---

I made it in the same key, it might. Let me test it on my side before

---

It’s valid and runs: https://github.com/echoix/grass/actions/runs/8794042045/job/24132847627